internal/conpty/conpty.go

```diff
@@ -0,0 +1,153 @@
+package conpty
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	errClosedConPty   = errors.New("pseudo console is closed")
+	errNotInitialized = errors.New("pseudo console hasn't been initialized")
+)
+
+// CPty is a wrapper around a Windows PseudoConsole handle. Create a new instance by calling `New()`.
+type CPty struct {
```

Comment:
can we call this something else? Is there already something named "ConPty"?

Reply:
No, was trying to avoid *conpty.ConPty but ehh. Changing

Reply:
Done

Reply:
Named ConPTY though